python/ray/experimental/collective/allreduce.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from typing import List, Optional, Union
+
+import ray
+from ray.dag.collective_node import CollectiveOutputNode, _CollectiveGroup
+from ray.dag.constants import (
+    BIND_INDEX_KEY,
+    COLLECTIVE_GROUP_KEY,
+    PARENT_CLASS_NODE_KEY,
+)
+from ray.experimental.channel.torch_tensor_type import GPUCommunicator, TorchTensorType
+from ray.experimental.util.types import ReduceOp
+from ray.util.collective import types as ray_collective_types
+
+logger = logging.getLogger(__name__)
+
+
+class AllReduceWrapper:
+    """Wrapper for NCCL all-reduce."""
+
+    def bind(
+        self,
+        input_nodes: List["ray.dag.DAGNode"],
+        op: ReduceOp = ReduceOp.SUM,
```

Comment:
Let's try to make it so that the `op` type is the same for `bind` vs `__call__` (see [other comment](https://github.com/ray-project/ray/pull/47621/files#r1800432289))

Reply:
Done.